File: app/src/main/java/com/example/model/TradingModels.kt
package com.example.model

enum class SignalType(val label: String, val binaryLabel: String, val isBullish: Boolean) {
  BUY("BUY / LONG", "CALL (UP)", true),
  SELL("SELL / SHORT", "PUT (DOWN)", false),
  CALL("CALL (UP)", "CALL (UP)", true),
  PUT("PUT (DOWN)", "PUT (DOWN)", false);

  val isCall: Boolean get() = isBullish
  val isBuy: Boolean get() = isBullish
}

data class MarketAsset(
  val id: String,
  val symbol: String,
  val name: String,
  val price: String,
  val changePercentage: String,
  val isBullish: Boolean,
  val sparklinePoints: List<Float>,
  val assetType: String = "Crypto",
  val volume24h: String = "$28.4B",
  val high24h: String = "",
  val low24h: String = "",
  val isOtc: Boolean = false,
  val otcPayout: String = "92%"
)

data class TradingSignal(
  val id: String,
  val pair: String,
  val type: SignalType,
  val timeframe: String = "1H",
  val entryPrice: String,
  val stopLoss: String = "",
  val takeProfit: String = "",
  val riskRewardRatio: String = "",
  val expiryTime: String = "1 Min Expiry",
  val confidencePercentage: Int = 92,
  val smcRationale: String = "Order Block Rejection",
  val timestamp: String = "Just now",
  val aiRationale: String,
  val payoutPercentage: String = "92%",
  val isOtc: Boolean = false,
  val category: String = "Crypto"
)

enum class DashboardTab(val title: String) {
  DASHBOARD("Dashboard"),
  SCANNER("AI Scanner"),
  SIGNALS("Signals"),
  ANALYTICS("Analytics"),
  PROFILE("Profile")
}

enum class BrokerType(
  val displayName: String,
  val shortName: String,
  val tagline: String,
  val defaultUrl: String,
  val latency: String,
  val primaryColorHex: Long,
  val accentColorHex: Long,
  val badgeText: String
) {
  QUOTEX(
    displayName = "Quotex",
    shortName = "QX",
    tagline = "Smart Binary & Digital Options",
    defaultUrl = "https://qxbroker.com",
    latency = "12ms",
    primaryColorHex = 0xFF00A3FF,
    accentColorHex = 0xFF00E676,
    badgeText = "QX PRO"
  ),
  POCKET_OPTION(
    displayName = "Pocket Option",
    shortName = "PO",
    tagline = "Quick Trading & High Yields",
    defaultUrl = "https://pocketoption.com",
    latency = "16ms",
    primaryColorHex = 0xFF2979FF,
    accentColorHex = 0xFFFFB800,
    badgeText = "PO LIVE"
  )
}

File: lib/models/trading_models.dart
import 'package:flutter/foundation.dart';

enum SignalType {
  buy('BUY / LONG', 'CALL (UP)', true),
  sell('SELL / SHORT', 'PUT (DOWN)', false),
  call('CALL (UP)', 'CALL (UP)', true),
  put('PUT (DOWN)', 'PUT (DOWN)', false);

  final String label;
  final String binaryLabel;
  final bool isBullish;

  const SignalType(this.label, this.binaryLabel, this.isBullish);
}

@immutable
class MarketAsset {
  final String id;
  final String symbol;
  final String name;
  final String price;
  final String changePercentage;
  final bool isBullish;
  final List<double> sparklinePoints;
  final String assetType;
  final String volume24h;
  final bool isOtc;
  final String otcPayout;

  const MarketAsset({
    required this.id,
    required this.symbol,
    required this.name,
    required this.price,
    required this.changePercentage,
    required this.isBullish,
    required this.sparklinePoints,
    this.assetType = 'Crypto',
    this.volume24h = '\$28.4B',
    this.isOtc = false,
    this.otcPayout = '92%',
  });
}

@immutable
class TradingSignal {
  final String id;
  final String pair;
  final SignalType type;
  final String timeframe;
  final String entryPrice;
  final String stopLoss;
  final String takeProfit;
  final String riskRewardRatio;
  final String expiryTime;
  final int confidencePercentage;
  final String smcRationale;
  final String timestamp;
  final String aiRationale;
  final String payoutPercentage;
  final bool isOtc;
  final String category;

  const TradingSignal({
    required this.id,
    required this.pair,
    required this.type,
    this.timeframe = '1H',
    required this.entryPrice,
    this.stopLoss = '',
    this.takeProfit = '',
    this.riskRewardRatio = '',
    this.expiryTime = '1 Min Expiry',
    this.confidencePercentage = 94,
    this.smcRationale = 'Order Block Rejection',
    this.timestamp = 'Just now',
    required this.aiRationale,
    this.payoutPercentage = '94%',
    this.isOtc = false,
    this.category = 'Forex',
  });
}
